core-utilities/src/main/java/edu/illinois/cs/cogcomp/core/datastructures/textannotation/Constituent.java

```diff
@@ -88,8 +94,22 @@ public Constituent(String label, String viewName, TextAnnotation text, int start
      * @param end end token offset (one-past-the-end)
      */
     public Constituent(String label, double score, String viewName, TextAnnotation text, int start,
-            int end) {
+                       int end) {
+        this(label, score, viewName, text, start, end, start, end, false);
+    }
 
+    public Constituent(String label, String viewName, TextAnnotation text, int start,
+                       int end, int headStart, int headEnd) {
+        this(label, 1.0, viewName, text, start, end, headStart, headEnd, true);
+    }
+
+    public Constituent(String label, double score, String viewName, TextAnnotation text, int start,
+                       int end, int headStart, int headEnd) {
+        this(label, score, viewName, text, start, end, headStart, headEnd, true);
+    }
+
+    private Constituent(String label, double score, String viewName, TextAnnotation text, int start,
+                       int end, int headStart, int headEnd, boolean hasHeadInformation) {
```

Comment:
Why not make `hasHeadInformation`, `private` and with default `false` value. Whenever you use the constructers which accept head information you make `hasHeadInformation`, `true`.